engine/runtime/environment.go

```diff
@@ -92,8 +92,15 @@ func (w *WasmEnvironment) UseInstance(ctx *scheduler.Ctx, instFunc func(*WasmIns
 	// return it to the environment when finished
 	inst := <-w.availableInstances
 
+	go func() {
+		if err := w.AddInstance(); err != nil {
+			panic(err)
```

Comment:
@cohix Not sure what we should do in this case. This always needs to succeed, else we get into a deadlock.

Reply:
Panic is really the only option, yeah.